### DISCUSSION TITLE
audio_test example -help  required

### DISCUSSION THREAD

**Discussion by petercwallis** [hathach/tinyusb#1608] [Q&A]
I have tried to add code to the audio_test example so that sound is detected by an MAX9814 mirophone from Adafruit connected to the adc on a rp2040 pico.  I think I have set up the tusb_config.h file and the usb_descriptors.c correctly to send 16kHz, 2 bytes per sample (unsigned) to the host. I remove the adc_read and replaced it with a constant value, and when I plug it into a machine running audacity.  I find the microphone and get data, but rather than a lovely straight line, I get a beautiful saw-tooth wave.

I guess I am using tud_audio_write(uint8_t *, unit16_t) wrongly.  Anyone got any pointers?  This could certainly be better documented.

Thanks in advance
p

**Comment by siddhartha993549** [comment]
Hi,

Can you please explain, What exactly is audio test example supposed to do ?
